types/react-loadable/index.d.ts

```diff
@@ -1,184 +1,190 @@
+         *
+         * ```ts
+         * Loadable({
+         *     // ...
+         *     render(loaded, props) {
+         *         const Component = loaded.default;
+         *         return <Component {...props} />
+         *     }
+         * });
+         * ```
+         */
+        render(loaded: Exports, props: Props): React.ReactNode;
+    }
+
+    interface LoadableComponent {
+        /**
+         * The generated component has a static method preload() for calling the loader function ahead of time.
+         * This is useful for scenarios where you think the user might do something next and want to load the
+         * next component eagerly.
+         *
+         * Note: preload() intentionally does not return a promise. You should not be depending on the timing of
+         * preload(). It's meant as a performance optimization, not for creating UI logic.
+         */
+        preload(): void;
+    }
+
+    interface LoadableCaptureProps {
+        /**
+         * Function called for every moduleName that is rendered via React Loadable.
+         */
+        report: (moduleName: string) => void;
+    }
+
+    interface Loadable {
+        <Props, Exports extends object>(options: Options<Props, Exports>): React.ComponentType<Props> & LoadableComponent;
+        Map<Props, Exports extends { [key: string]: any }>(options: OptionsWithMap<Props, Exports>): React.ComponentType<Props> & LoadableComponent;
+
+        /**
+         * This will call all of the LoadableComponent.preload methods recursively until they are all
+         * resolved. Allowing you to preload all of your dynamic modules in environments like the server.
+         * ```ts
+         * Loadable.preloadAll().then(() => {
+         *   app.listen(3000, () => {
+         *     console.log('Running on http://localhost:3000/');
+         *   });
+         * });
+         * ```
+         */
+        preloadAll(): Promise<void>;
+
+        /**
+         * Check for modules that are already loaded in the browser and call the matching
+         * `LoadableComponent.preload` methods.
+         * ```ts
+         * window.main = () => {
+         *   Loadable.preloadReady().then(() => {
+         *     ReactDOM.hydrate(
+         *       <App/>,
+         *       document.getElementById('app'),
+         *     );
+         *   });
+         * };
+         * ```
+         */
+        preloadReady(): Promise<void>;
+
+        Capture: React.ComponentType<LoadableCaptureProps>;
+    }
 }
 
-export interface OptionsWithoutRender<Props> extends CommonOptions {
-    /**
-     * Function returning a promise which returns a React component displayed on success.
-     *
-     * Resulting React component receives all the props passed to the generated component.
-     */
-    loader(): Promise<React.ComponentType<Props> | { default: React.ComponentType<Props> }>;
-}
+declare const LoadableExport: Loadable.Loadable;
```

Comment:
Same, this creates a `LoadableExport` global that tsc would accept. Needs to be inside the `declare module` block.

Reply:
I believe this issue with this is that you cannot export multiple "things" when using module assignment.